protocol-tests/swagger/required-optional-body.json

```diff
@@ -0,0 +1,134 @@
+{
```

Comment:
Does this test provide more scenario than other existing tests (testserver and others), or can it be combined with existing?

We might not want to add too many test swagger, as it have some cost on dev time and CI time.

Reply:
It just contains different cases in one swagger json so that it's easier to validate the result. But autorest.testserver covers those cases as well in different swagger json. We can remove this json.